sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/CertificateClientImpl.java

```diff
@@ -0,0 +1,1771 @@
+                status = LongRunningOperationStatus.IN_PROGRESS;
+                break;
+            case "completed":
+                status = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
+                break;
+            case "failed":
+                status = LongRunningOperationStatus.FAILED;
+                break;
+            default:
+                //should not reach here
+                status = LongRunningOperationStatus.fromString(certificateOperationResponse.getValue().getStatus(), true);
+                break;
+        }
+        return Mono.just(new PollResponse<>(status, certificateOperationResponse.getValue()));
+    }
+
+    public Mono<Response<CertificateOperation>> cancelCertificateOperationWithResponseAsync(String certificateName, Context context) {
+        context = context == null ? Context.NONE : context;
+        CertificateOperationUpdateParameter parameter = new CertificateOperationUpdateParameter().cancellationRequested(true);
+
+        return service.updateCertificateOperationAsync(vaultUrl, certificateName, apiVersion, ACCEPT_LANGUAGE, parameter, CONTENT_TYPE_HEADER_VALUE,
+                context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE))
+            .doOnRequest(ignored -> logger.verbose("Cancelling certificate operation - {}",  certificateName))
+            .doOnSuccess(response -> logger.verbose("Cancelled the certificate operation - {}", response.getValue().getStatus()))
+            .doOnError(error -> logger.warning("Failed to cancel the certificate operation - {}", certificateName, error));
+    }
+
+    public Response<CertificateOperation> cancelCertificateOperationWithResponse(String certificateName, Context context) {
+        context = context == null ? Context.NONE : context;
+        context = enableSyncRestProxy(context);
+        CertificateOperationUpdateParameter parameter = new CertificateOperationUpdateParameter().cancellationRequested(true);
+        return service.updateCertificateOperation(vaultUrl, certificateName, apiVersion, ACCEPT_LANGUAGE, parameter, CONTENT_TYPE_HEADER_VALUE,
+                context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE));
+    }
+
+    private Mono<Response<CertificateOperation>> createCertificateWithResponseAsync(String certificateName, CertificatePolicy certificatePolicy, boolean enabled, Map<String, String> tags, Context context) {
+        CertificateRequestParameters certificateRequestParameters = new CertificateRequestParameters()
+            .certificatePolicy(new CertificatePolicyRequest(certificatePolicy))
+            .certificateAttributes(new CertificateRequestAttributes().enabled(enabled))
+            .tags(tags);
+        return service.createCertificateAsync(vaultUrl, certificateName, apiVersion, ACCEPT_LANGUAGE, certificateRequestParameters, CONTENT_TYPE_HEADER_VALUE,
+            context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE));
```

Comment:
I noticed we didn't have logging statements for this operation before, but I think we should:
```suggestion
        return service.createCertificateAsync(vaultUrl, certificateName, apiVersion, ACCEPT_LANGUAGE, certificateRequestParameters, CONTENT_TYPE_HEADER_VALUE,
            context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE))
            .doOnRequest(ignored -> logger.verbose("Starting creation of certificate - {}",  certificateName))
            .doOnSuccess(response -> logger.verbose("Started creation of certificate - {}", certificateName))
            .doOnError(error -> logger.warning("Failed to create the certificate - {}", certificateName, error));
```